src/main/jbake/content/blog/2017/02/13/monitoring-canary-deployments.adoc

```diff
@@ -0,0 +1,249 @@
+:jbake-status: published
+:jbake-tags: blog, apm, opentracing, vertx, openshift
+
+In our last release, link:/blog/2017/02/04/hawkular-apm-service-deployments.html[we built a Services screen]
+on top of a feature added a couple of releases ago, the "service name" and "build
+stamps". These properties can be specified as an environment variable or system
+property, and are automatically detected when the monitored application runs on
+OpenShift.
+
+On this blog post, we'll show you how to use that screen to monitor a Canary release
+scenario. https://martinfowler.com/bliki/CanaryRelease.html[A Canary] release
+is a technique used to minimize downtime (and risk) between deployments by preparing
+the new version on a completely different production environment and, once this new
+version is ready to serve requests, redirect _some_ traffic from the "old" version
+to the "new" version. If this new version looks good, it gradually replaces the
+old version, eventually taking over all traffic. This technique builds on top of
+the https://martinfowler.com/bliki/BlueGreenDeployment.html[Blue-Green] deployment
+technique.
+
+We'll use our https://github.com/hawkular/hawkular-apm/tree/master/examples/vertx-opentracing[Vert.x OpenTracing example]
+for this demo. First, let's deploy the example into OpenShift, along with Hawkular APM,
+by running the following Ansible playbook:
+
+```bash
+$ ansible-playbook vertx-opentracing.yml
+```
+
+After a few minutes, it should print out the address for the Order Manager microservice,
+like this:
+
+```
+TASK [Print out the order-manager address] *************************************
+ok: [localhost] => {
+    "msg": "The hostname for the Order Manager service is: http://order-manager-order-manager.192.168.2.107.xip.io"
+}
+```
+
+At this point, change the `genorders.sh` script, to use this host instead of the
+original `http://localhost:8180` on the `curl` command. It should then read:
+`curl -X POST -H "Content-Type: application/json" -d "$data" http://order-manager-order-manager.192.168.2.107.xip.io/orders`.
+Note that your hostname might be different than our example. You can then run
+the `genorders.sh` script, which will simulate some random requests to the Order
+Manager. The script will execute until it's cancelled with `Ctrl+C`, and should
+have an output similar to this:
+
+```
+Creating an order for account=brian item=laptop quantity=3
+{
+  "accountId" : "brian",
+  "itemId" : "laptop",
+  "quantity" : 3,
+  "id" : "d3f09a97-6f87-45c5-98c8-b592b6beb501"
+}
+
+Creating an order for account=sarah item=car quantity=4
+Not account found
+
+Creating an order for account=steve item=car quantity=2
+{
+  "accountId" : "steve",
+  "itemId" : "car",
+  "quantity" : 2,
+  "id" : "7f095b1e-94a2-4398-8480-01aaea43e021"
+}
+...
+...
+```
+
+Hawkular APM should be located under the service name `hawkular-apm`, so, the URL
+be similar to this: `https://hawkular-apm-order-manager.192.168.2.107.xip.io`. Note
```

Comment:
No, wait, it's correct: the Ansible playbook deploys Hawkular APM on the Order Manager project, so, the URL is correct: `servicename-projectname.ip.xip.io` .